src/client/ui/controls.js

```diff
@@ -138,6 +204,21 @@ export default class StorybookControls extends React.Component {
         <div style={h1WrapStyle}>
           <h3 style={h1Style}>React Storybook</h3>
         </div>
+        <div style={filterTextWrapStyle}>
+          <input
+            style={filterTextStyle}
+            type="text"
+            placeholder="Filter"
+            name="filter-text"
+            value={this.state.filterText}
+            onChange={this.filterStoryList.bind(this)}
```

Comment:
do  `this.filterStoryList=this.filterStoryList.bind(this)` in the constructor . Here it will create a different listener on every re-render.